assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py

```diff
@@ -270,17 +273,25 @@ def run(args):
         log_time_and_message("Reading data in spark and converting to pandas")
         baseline_df = try_read_mltable_in_spark_with_error(args.baseline_data, "baseline_data")
 
+        categorical_features = get_categorical_cols_with_df_with_override(baseline_df,
+                                                                              args.override_numerical_features,
+                                                                              args.override_categorical_features)
         baseline_df = baseline_df.toPandas()
-        task_type = determine_task_type(args.task_type, args.target_column, baseline_df)
+        # lightgbm does not support features that can't be converted to non-bool, int, float features
+        # if these features can't be converted to an int. Therefore we have to more these as "category"
+        # types
+        categorical_features_lgbm = compute_lightgbm_unsupported_categorical_features(baseline_df,
```

Comment:
also, I suppose this compute_feature_importance component is not lightgbm specific? why apply lightgbm speicifc logic here?

Reply:
According to Hannah, this is the current limitation. Now we only use the lightgbm model for the feature importance calculation. Ideally we'd use the model that the user is using